extensions/gc/GarbageCollectorConfigValidator.js

```diff
@@ -13,6 +13,7 @@ const joiSchema = joi.object({
         groupId: joi.string().required(),
         retry: retryParamsJoi,
         concurrency: joi.number().greater(0).default(10),
+        maxQueued: joi.number().greater(0).default(1000),
```

Comment:
regarding the config, I think we should double check where (which extension) this feature should be used?

i.e. we probably don't need it everywhere, so best be explicit and add the `maxQueued` parameter only in the extensions where it makes sense (i.e. there is indeed a `key`), and set the `orderByFunc` to null otherwise.

Reply:
Removed from The Bucket Processor as it's not needed, and the GC as well as the messages it receives don't have a key.